.travis.yml

```diff
@@ -7,12 +7,19 @@ script:
     - docker run my-test-image tests/unit
     - docker run my-test-image tests/integration
     - docker run --env FLASK_ENV --env BOARD_ID --env MONGO_CONNECTION_URL my-test-image tests/endtoend
+install:
+    - wget https://releases.hashicorp.com/terraform/"$TF_VERSION"/terraform_"$TF_VERSION"_linux_amd64.zip
+    - unzip terraform_"$TF_VERSION"_linux_amd64.zip
+    - sudo mv terraform /usr/local/bin/
+    - rm terraform_"$TF_VERSION"_linux_amd64.zip
+    - terraform init
 before_deploy:
     - docker build --target production --tag $DOCKER_USER/todo-app:latest .  
     - docker push $DOCKER_USER/todo-app:latest
+    - terraform apply -auto-approve
 deploy:   
     provider: script
     script: 
-        - curl -dH -X POST "$AZURE_WEBHOOK_URL"
+        - curl -dH -X POST "$(terraform output -raw cd_webhook)"
```

Comment:
Worth noting that this is erroring in your Travis build currently - you might find it worthwhile adding a `--fail` flag to curl to ensure that it returns an error code if the status code it receives isn't a success.
I can't see a clear reason for that error here, but it could be because of the special characters; I'd suggest trying either  manually adding a `\` to escape the $ in your cd_webhook output, or moving this command into a bash script that's called here as that can subtly change the behaviour

Reply:
Hi Jack,
I have made the changes and pushed to my branch. The Travis build (deploy step) fails with this error :22.38s$ terraform apply -var 'CLIENT_ID=$CLIENT_ID' -var 'CLIENT_SECRET=$CLIENT_SECRET' -var 'DOCKER_PASSWORD=$DOCKER_PASSWORD' -auto-approve
dpl_0
1.48s$ rvm $(travis_internal_ruby) --fuzzy do ruby -S gem install dpl
dpl.1
Installing deploy dependencies
dpl.2
Preparing deploy
dpl.3
Deploying application
HEAD detached at a0a01a4
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   .terraform.lock.hcl
no changes added to commit (use "git add" and/or "git commit -a")
Dropped refs/stash@{0} (60540db2bc3083d1e69ef847a81816b47de901d7)
Script failed with status 22
failed to deploy

Reply:
I am not sure what is causing this. Could you please let me know ?